hack/tools/extractcheck/extractcheck.go

```diff
@@ -0,0 +1,154 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"go/ast"
+	"go/types"
+	"path"
+	"regexp"
+	"strings"
+
+	"golang.org/x/tools/go/analysis"
+)
+
+type config struct {
+	matches []*regexp.Regexp
+}
+
+func newConfig() config {
+	var matches []*regexp.Regexp
+	filterRegex := []string{
+		"k8s.io/kubernetes/cmd/kube-controller-manager/.*",
+		"k8s.io/kubernetes/cmd/kube-scheduler/.*",
+		"k8s.io/kubernetes/pkg/controller/.*",
+		"k8s.io/kubernetes/pkg/scheduler/.*",
```

Comment:
How do we know this set is good enough?

Reply:
We don't. This is a list I can think of that is relevant to KCM and scheduler. Alternatively, we can just simply check the whole code base except for the e2e tests, which afaik is the only the only place using `Extract`. WDTY?

Reply:
Let's start here.

Reply:
Changed it to excludeRegex

Reply:
I prefer what you suggested as alternative ("we can just simply check the whole code base except for the e2e tests").

The inverted path match from https://github.com/golangci/golangci-lint/pull/3617 can be used to suppress issues in tests where the functions are still uses legitimately.